crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -411,6 +411,35 @@ impl Query {
         .extend()
     }
 
+    /// The Move packages that exist in the network, optionally filtered to be strictly before
+    /// `beforeCheckpoint` and/or strictly after `afterCheckpoint`.
+    ///
+    /// This query will return all versions of a given user package that appear between the
+    /// specified checkpoints, but only records the latest versions of system packages.
+    async fn packages(
+        &self,
+        ctx: &Context<'_>,
+        first: Option<u64>,
+        after: Option<move_package::Cursor>,
+        last: Option<u64>,
+        before: Option<move_package::Cursor>,
+        after_checkpoint: Option<u64>,
+        before_checkpoint: Option<u64>,
```

Comment:
bikeshedding: do we want to tuck any filtering criteria in a `filter` param?

Reply:
I don't feel super strongly about it. The reason I didn't do it initially is that I'm introducing two different pagination queries for packages (by checkpoint and by version) and I didn't want to sew confusion by having the two different filter types, but I'm curious to hear what @hayes-mysten has to say on GraphQL ergonomics and what is easier to use in practice.

Reply:
Not sure if I follow here, but I think having unique filters for different fields isn't a big deal for query ergonomics, but I'm not sure what the filters would be/how they would be different between the 2 query fields

Reply:
I think I get it now, I don't have a strong preference between the 2 options, but wrapping in a filter would be more consistent with how the other filters work.  Is the issue that we don't want to use checkpoint and versioning filters at the same time?

Reply:
Yes that's right, so if we want to keep these two kinds of filters separate, we would need to introduce two different input types: `MovePackageCheckpointFilter` and `MovePackageVersionFilter` each with just a pair of numbers in, so it started to feel a bit overkill, but I buy a consistency argument.